src/jace/util/compiling.py

```diff
@@ -0,0 +1,146 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Contains everything for compiling and running `TranslatedJaxprSDFG` instances."""
+
+from __future__ import annotations
+
+import time
+from typing import TYPE_CHECKING, Any
+
+import dace
+import numpy as np
+from dace import data as dace_data
+
+from jace import util
+
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence
+
+    from jace import translator
+    from jace.util import dace_helper
+
+
+def compile_jax_sdfg(
+    tsdfg: translator.TranslatedJaxprSDFG,
+) -> dace_helper.CompiledSDFG:
+    """Compiles the SDFG embedded in `tsdfg` and return the resulting `CompiledSDFG` object.
+
+    Note:
+        For calling the returned `CompiledSDFG` object you need the `inp_names` and `out_names`
+        of the input `TranslatedJaxprSDFG`.
+    """
+    if any(  # We do not support the DaCe return mechanism
+        arrname.startswith("__return")
+        for arrname in tsdfg.sdfg.arrays.keys()  # noqa: SIM118  # we can not use `in` because we are also interested in `__return_`!
```

Comment:
Rename `arrname` to either `arr_name` (ugly) or just `name` to follow coding conventions.